benchmarks/benchmarks/stats.py

```diff
@@ -56,7 +56,99 @@ def time_ttest_ind_diff_var(self):
+    ]
+    dist_data = dict(distcont + distdiscrete)
+    # custom shape values can be provided for any distribution in the format
+    # `dist_name`: [shape1, shape2, ...]
+    custom_input = {}
+
+    # these are the distributions that are the slowest
+    slow_dists = ['nct', 'ncx2', 'argus', 'cosine', 'foldnorm', 'gausshyper',
+                  'kappa4', 'invgauss', 'wald', 'vonmises_line', 'ksone',
+                  'genexpon', 'exponnorm', 'recipinvgauss', 'vonmises',
+                  'foldcauchy', 'kstwo', 'levy_stable', 'skewnorm']
+    slow_methods = ['moment']
+
+    def setup(self, distribution, properties):
+        if not is_xslow() and (distribution in self.slow_dists or
+                               properties in self.slow_methods):
+            raise NotImplementedError("Skipped")
+
+        self.dist = getattr(stats, distribution)
+
+        shapes = self.dist_data[distribution]
+
+        if isinstance(self.dist, stats.rv_discrete):
+            # discrete distributions only use location
+            self.isCont = False
+            kwds = {'loc': 4}
+        else:
+            # continuous distributions use location and scale
+            self.isCont = True
+            kwds = {'loc': 4, 'scale': 10}
+
+        bounds = self.dist.interval(.99, *shapes, **kwds)
+        x = np.linspace(*bounds, 100)
+        args = [x, *self.custom_input.get(distribution, shapes)]
+
+        if properties == 'fit':
+            # the only positional argument is the data to be fitted
+            self.args = self.dist.rvs(size=100, **kwds)
+        elif properties == 'rvs':
+            # add size keyword argument for data creation
+            kwds['size'] = 1000
+            # keep shapes as positional arguments, omit linearly spaced data
+            self.args = args[1:]
+        elif properties == 'pdf/pmf':
+            properties = ('pmf' if isinstance(self.dist, stats.rv_discrete)
+                          else 'pdf')
+            self.args = args
+        elif properties == 'logpdf/logpmf':
+            properties = ('logpmf' if isinstance(self.dist, stats.rv_discrete)
+                          else 'logpdf')
+            self.args = args
+        elif properties in ['ppf', 'isf']:
+            self.args = [np.linspace((0, 1), 100), *args[1:]]
+        elif properties == 'moment':
+            # the first four moments may be optimized, so compute the fifth
+            self.args = [5, *args[1:]]
+        elif properties.startswith('stats_'):
+            properties = 'stats'
+            self.args = args[1:]
+            kwds['moments'] = properties[6:]
+            self.kwds = kwds
+        elif properties == 'entropy':
+            self.args = args[1:]
+        else:
+            self.args = args
+        self.kwds = kwds
+
+        try:
+            self.method = getattr(self.dist, properties)
+        except AttributeError:
+            raise NotImplementedError("This attribute is not a member "
```

Comment:
Does this ever happen?

Reply:
It does occur - discrete distributions do not have a fit method. 

Though I thought to manually make sure that nothing fishy is going on in the logic, and noticed that even though some discrete distributions are clearly listed as discrete, meaning they need the pmf and logpmf, they don't appear to be an instance of the rv_discrete class. 


```
isinstance(stats.skellam, stats.rv_discrete)
Out[30]: False 
```

Reply:
Umm... on my computer this example evaluates to `True`.

Reply:
strange. I copied that directly over from the spyder's ipython, but I now also see that it is true.